lib/identity_cache/replicated_cache_proxy.rb

```diff
@@ -0,0 +1,46 @@
+# frozen_string_literal: true
+
+module IdentityCache
+  class ReplicatedCacheProxy
+    def initialize(cache_backend, blob_replication_factor:)
+      @cache_backend = cache_backend
+      @blob_replication_factor = blob_replication_factor
+    end
+
+    def write(key, value)
+      keys(key).each do |key|
+        cache_backend.write(key, value)
+      end
+    end
+
+    def delete(key)
+      keys(key).each do |key|
```

Comment:
I think the code becomes harder to read by having "key" meaning two different things.  We could rename the block argument to `replica_key` to distinguish between the two types of keys.  Also, we could have an `each_replica_key` method that yields the keys without unnecessary creating an array.

  E.g.

```ruby
    def delete(key, value)
      each_replica_key(key) do |replica_key|
```